frontend/tests/util/blocks.ts

```diff
@@ -0,0 +1,238 @@
+                const input = page.getByText(block.query);
+                await input.pressSequentially(newSeries.title);
+                await page.getByRole("option", { name: newSeries.title }).click();
+                await saveButton.click();
+
+                await expect(page.getByRole("heading", { name: newSeries.title })).toBeVisible();
+            });
+
+            await test.step("Video order can be changed", async () => {
+                await editButton.click();
+                await page.getByLabel("Oldest first").setChecked(true);
+                await expect(page.getByLabel("Oldest first")).toBeChecked();
+                await saveButton.click();
+
+                await expect(
+                    page.getByRole("button", { name: "Choose video order" })
+                ).toHaveText("Oldest first");
+            });
+
+            await test.step("Layout options can be changed", async () => {
+                await test.step("Title and videos (initial)", async () => {
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.title })
+                    ).toBeVisible();
+                });
+
+                await test.step("Videos only", async () => {
+                    await editButton.click();
+                    await page.getByLabel("Show title").setChecked(false);
+                    await page.getByLabel("Show description").setChecked(false);
+                    await saveButton.click();
+
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.title })
+                    ).not.toBeVisible();
+                    await expect(
+                        page.getByText("Cat software running on dog hardware.")
+                    ).not.toBeVisible();
+                });
+
+                await test.step("Description and videos", async () => {
+                    await editButton.click();
+                    await page.getByLabel("Show description").setChecked(true);
+                    await saveButton.click();
+
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.description })
+                    ).not.toBeVisible();
+                    await expect(page.getByText(newSeries.description).first()).toBeVisible();
+                });
+
+                await test.step("Title, description and videos", async () => {
+                    await editButton.click();
+                    await page.getByLabel("Show title").setChecked(true);
+                    await page.getByLabel("Show description").setChecked(true);
+                    await saveButton.click();
+
+                    await expect(
+                        page.getByRole("heading", { name: newSeries.title })
+                    ).toBeVisible();
+                    await expect(page.getByText(newSeries.description).first()).toBeVisible();
```

Comment:
The `.first()` and `toBeVisible` together seem fishy. If the `first()` is necessary, then there is presumably a second element with that text somewhere. In that case, the assertion `first().not.toBeVisible` is useless. So yeah, this exact line (without the `not`) is fine. But above it might be wrong. Wait, in the `not.toBeVisible` case there is no `first()`. Why do we need it here then?

Reply:
I also left a code comment explaining why I _believe_ that this is necessary.